bundles/admin/admin-layereditor/view/AdminLayerForm/VisualizationTabPane/VectorStyle/VectorStyleSelect.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { List, Checkbox } from 'antd';
+import { EditOutlined, DeleteOutlined } from '@ant-design/icons';
+import { Button } from 'oskari-ui';
+import { Controller } from 'oskari-ui/util';
+import styled from 'styled-components';
+
+const StyledItem = styled(List.Item)`
+    & {
+        background: #ffffff;
+        line-height: 30px;
+        padding: 0 11px;
+    }
+`;
+
+export const VectorStyleSelect = (props) => {
+    const styles = props.layer.options.styles;
+
+    if (!styles && typeof styles === 'undefined') {
+        return null;
+    }
+
+    return (
+        <List
+            dataSource={ Object.keys(props.layer.options.styles).sort((a, b) => Oskari.util.naturalSort(a, b)) }
```

Comment:
The sort() method sorts the elements of an array _in place_ and returns the sorted array. We don't want to modify the state so you should slice() before calling sort().

Reply:
See https://github.com/kasparia/oskari-frontend/pull/2